gen/fonts.c

```diff
@@ -1,268 +1,18 @@
-	/* 0xb6 _ */ (uint8_t *)"\x01\x00",
-	/* 0xb7 _ */ (uint8_t *)"\x01\x00",
-	/* 0xb8 _ */ (uint8_t *)"\x01\x00",
-	/* 0xb9 _ */ (uint8_t *)"\x01\x00",
-	/* 0xba _ */ (uint8_t *)"\x01\x00",
-	/* 0xbb _ */ (uint8_t *)"\x01\x00",
-	/* 0xbc _ */ (uint8_t *)"\x01\x00",
-	/* 0xbd _ */ (uint8_t *)"\x01\x00",
-	/* 0xbe _ */ (uint8_t *)"\x01\x00",
-	/* 0xbf _ */ (uint8_t *)"\x01\x00",
-	/* 0xc0 _ */ (uint8_t *)"\x01\x00",
-	/* 0xc1 _ */ (uint8_t *)"\x01\x00",
-	/* 0xc2 _ */ (uint8_t *)"\x01\x00",
-	/* 0xc3 _ */ (uint8_t *)"\x01\x00",
-	/* 0xc4 _ */ (uint8_t *)"\x01\x00",
-	/* 0xc5 _ */ (uint8_t *)"\x01\x00",
-	/* 0xc6 _ */ (uint8_t *)"\x01\x00",
-	/* 0xc7 _ */ (uint8_t *)"\x01\x00",
-	/* 0xc8 _ */ (uint8_t *)"\x01\x00",
-	/* 0xc9 _ */ (uint8_t *)"\x01\x00",
-	/* 0xca _ */ (uint8_t *)"\x01\x00",
-	/* 0xcb _ */ (uint8_t *)"\x01\x00",
-	/* 0xcc _ */ (uint8_t *)"\x01\x00",
-	/* 0xcd _ */ (uint8_t *)"\x01\x00",
-	/* 0xce _ */ (uint8_t *)"\x01\x00",
-	/* 0xcf _ */ (uint8_t *)"\x01\x00",
-	/* 0xd0 _ */ (uint8_t *)"\x01\x00",
-	/* 0xd1 _ */ (uint8_t *)"\x01\x00",
-	/* 0xd2 _ */ (uint8_t *)"\x01\x00",
-	/* 0xd3 _ */ (uint8_t *)"\x01\x00",
-	/* 0xd4 _ */ (uint8_t *)"\x01\x00",
-	/* 0xd5 _ */ (uint8_t *)"\x01\x00",
-	/* 0xd6 _ */ (uint8_t *)"\x01\x00",
-	/* 0xd7 _ */ (uint8_t *)"\x01\x00",
-	/* 0xd8 _ */ (uint8_t *)"\x01\x00",
-	/* 0xd9 _ */ (uint8_t *)"\x01\x00",
-	/* 0xda _ */ (uint8_t *)"\x01\x00",
-	/* 0xdb _ */ (uint8_t *)"\x01\x00",
-	/* 0xdc _ */ (uint8_t *)"\x01\x00",
-	/* 0xdd _ */ (uint8_t *)"\x01\x00",
-	/* 0xde _ */ (uint8_t *)"\x01\x00",
-	/* 0xdf _ */ (uint8_t *)"\x01\x00",
-	/* 0xe0 _ */ (uint8_t *)"\x01\x00",
-	/* 0xe1 _ */ (uint8_t *)"\x01\x00",
-	/* 0xe2 _ */ (uint8_t *)"\x01\x00",
-	/* 0xe3 _ */ (uint8_t *)"\x01\x00",
-	/* 0xe4 _ */ (uint8_t *)"\x01\x00",
-	/* 0xe5 _ */ (uint8_t *)"\x01\x00",
-	/* 0xe6 _ */ (uint8_t *)"\x01\x00",
-	/* 0xe7 _ */ (uint8_t *)"\x01\x00",
-	/* 0xe8 _ */ (uint8_t *)"\x01\x00",
-	/* 0xe9 _ */ (uint8_t *)"\x01\x00",
-	/* 0xea _ */ (uint8_t *)"\x01\x00",
-	/* 0xeb _ */ (uint8_t *)"\x01\x00",
-	/* 0xec _ */ (uint8_t *)"\x01\x00",
-	/* 0xed _ */ (uint8_t *)"\x01\x00",
-	/* 0xee _ */ (uint8_t *)"\x01\x00",
-	/* 0xef _ */ (uint8_t *)"\x01\x00",
-	/* 0xf0 _ */ (uint8_t *)"\x01\x00",
-	/* 0xf1 _ */ (uint8_t *)"\x01\x00",
-	/* 0xf2 _ */ (uint8_t *)"\x01\x00",
-	/* 0xf3 _ */ (uint8_t *)"\x01\x00",
-	/* 0xf4 _ */ (uint8_t *)"\x01\x00",
-	/* 0xf5 _ */ (uint8_t *)"\x01\x00",
-	/* 0xf6 _ */ (uint8_t *)"\x01\x00",
-	/* 0xf7 _ */ (uint8_t *)"\x01\x00",
-	/* 0xf8 _ */ (uint8_t *)"\x01\x00",
-	/* 0xf9 _ */ (uint8_t *)"\x01\x00",
-	/* 0xfa _ */ (uint8_t *)"\x01\x00",
-	/* 0xfb _ */ (uint8_t *)"\x01\x00",
-	/* 0xfc _ */ (uint8_t *)"\x01\x00",
-	/* 0xfd _ */ (uint8_t *)"\x01\x00",
-	/* 0xfe _ */ (uint8_t *)"\x01\x00",
-	/* 0xff _ */ (uint8_t *)"\x01\x00",
+const uint8_t * const font_data[2][128] = {
+	{
+#include"font.inc"
+	},
+	{
+#include"fontfixed.inc"
+	},
 };
 
-int fontCharWidth(char c) {
-	return font_data[(int)(c)][0];
```

Comment:
BTW `(int)(c)` is broken, since char is signed.  However, `oledConvertChar` in oled.c made sure that there was no array out of bounds.